libsrc/osic1p/cgetc.s

```diff
@@ -8,8 +10,21 @@
         .include        "extzp.inc"
         .include        "zeropage.inc"
 
+; Initialize one-character buffer that is filled by kbhit()
+initcgetc:
+        lda     #$00
+        sta     CHARBUF         ; No character in buffer initially
+        rts

```

Comment:
I don't want to say that it would be preferable beause every access would be both longer and slower but nevertheless I'd like to point out that moving `CHARBUF` from zeropage to the BSS segment would make this constructor obsolete as the BSS segment is completely initialized to zero by the startup code in ctr0.s.

BTW: A variable that needs an initial value different from zero can be placed in the DATA segment. Then again there's no need for explicit initialization code as the value is read from the binary (just like code).